tensorflow/python/keras/layers/convolutional_test.py

```diff
@@ -257,6 +268,37 @@ def test_conv3d_dynamic_shape(self):
             input_shape=(None, 3, None, None, None),
             input_data=input_data)
 
+class GroupedConvTest(keras_parameterized.TestCase):
+  @parameterized.named_parameters(
+      ('Conv1D', keras.layers.Conv1D),
+      ('Conv2D', keras.layers.Conv2D),
+      ('Conv3D', keras.layers.Conv3D),
+  )
+  def test_group_conv_incorrect_use(self, layer):
+    with self.assertRaisesRegexp(ValueError, 'The number of filters'):
+      layer(16, 3, groups=3)
+    with self.assertRaisesRegexp(ValueError, 'The number of input channels'):
+      layer(16, 3, groups=4).build((32, 12, 12, 3))
+
+  @parameterized.named_parameters(
+      ('Conv1D', keras.layers.Conv1D, (32, 12, 32)),
+      ('Conv2D', keras.layers.Conv2D, (32, 12, 12, 32)),
+      ('Conv3D', keras.layers.Conv3D, (32, 12, 12, 12, 32)),
+  )
+  def test_group_conv(self, layer, input_shape):
+    if test.is_gpu_available(cuda_only=True):
+      with self.cached_session(use_gpu=True):
```

Comment:
I think we are using the `with test_util.use_gpu():` these days

Reply:
:+1: I didn't know that, most of the tests in this file still use `cached_session`. Changed it in f3d9db564a7190c6c33e2c5f8747be631b1f6b8d